metricbeat/module/beats/beats_integration_test.go

```diff
@@ -0,0 +1,92 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+// +build integration
+
+package beats_test
+
+import (
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/elastic/beats/libbeat/tests/compose"
+	mbtest "github.com/elastic/beats/metricbeat/mb/testing"
+	"github.com/elastic/beats/metricbeat/module/beats"
+	_ "github.com/elastic/beats/metricbeat/module/beats/stats"
+)
+
+var metricSets = []string{
+	"stats",
+}
+
+func TestFetch(t *testing.T) {
+	compose.EnsureUp(t, "beats")
+
+	for _, metricSet := range metricSets {
+		f := mbtest.NewReportingMetricSetV2Error(t, beats.GetConfig(metricSet))
+		events, errs := mbtest.ReportingFetchV2Error(f)
+
+		assert.Empty(t, errs)
+		if !assert.NotEmpty(t, events) {
+			t.FailNow()
+		}
+
+		t.Logf("%s/%s event: %+v", f.Module().Name(), f.Name(),
+			events[0].BeatEvent("beats", metricSet).Fields.StringToPrint())
+	}
+}
+
+func TestData(t *testing.T) {
```

Comment:
Why not use the new testing mechanism here with the golden files?

Reply:
Sorry, I think I already forgot about this 😞. Could you point me to an example module that's using this?

Reply:
this should help: https://github.com/elastic/beats/tree/master/metricbeat/mb/testing/data

Reply:
Okay, I started to use this new `testdata` framework in 627ecf2bb. However, as you will see the generated `stats.800.json-expected.json` file is bad. It has a `error.message` field with value as `"HTTP error 404 in : 404 Not Found"`.

Can you help me figure out what I'm doing wrong? Better yet, can you help me figure out how to debug what's going on with this new framework?

Reply:
Okay, I did some digging and I think I know why this is happening. The `beats/stats` metricset makes 2 HTTP API calls: `GET /` and `GET /stats`.

I'm investigating now how to mock both these calls with the `testdata` framework.

Reply:
AFAICT, there's no way currently for the `testdata` framework to handle the scenario where a metricset needs to call multiple HTTP APIs. Can you confirm this?

If this is true, I'd rather not hold up this PR (and the one coming after it for the `beat/state` metricset: https://github.com/elastic/beats/pull/12615) for this enhancement to be implemented in the framework. Once the enhancement is there I can make a follow up PR to use it with these metricsets. WDYT?

Reply:
I can confirm unfortunately, see https://github.com/elastic/beats/issues/11425 SGTM to move forward.